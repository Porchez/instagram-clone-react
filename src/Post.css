.post{
    max-width: 600px;
    background-color:  white;
    border: 1px solid lightgrey;
    margin-bottom: 15px;
}

@media (min-width: 620px) {
    .post{
        margin-bottom: 40px;
    }    
}

.post__image{
    width: 100%;
    object-fit: contain;
    border-top: 1px solid lightgrey;
    border-bottom: 1px solid lightgrey;
}

.post__header{
    display: flex;
    align-items: center;
    padding: 16px;
    padding-right: 40px;
}

.post__avatar{
    margin-right: 10px;
}

.post__capture{
    font-weight: normal;
    padding: 16px;
}

.post__input{
    display: flex;
    flex-direction: row;
    border-top: 1px solid lightgray !important;
}

.post__input__comment{
    border: none !important;
    
    border-radius: 0 !important;
}

.post__input__comment.form-control:focus {
    color: #495057;
    background-color: #fff;
    outline: 0;
    box-shadow:none;
}

.post__btn{
    color: #0095f6 !important;
    border: none !important;
    /* border-top: 1px solid lightgray !important; */
    border-radius: 0 !important;
    background-color: transparent !important;
    font-weight: 500 !important;
}

.post__comments{
    padding: 16px;
}