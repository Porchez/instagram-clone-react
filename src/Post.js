import React, { useState, useEffect } from 'react'
import './Post.css'
import { Avatar } from '@material-ui/core'
import { db } from './firebase';
import { Button, Form } from 'react-bootstrap';
import firebase from "firebase";

function Post({ postId, user, username, caption, imageUrl, photoURL }) {
    const [comment, setComment] = useState('');
    const [comments, setComments] = useState([]);

    useEffect(() => {
        let unsubscribe;
        if (postId) {
            unsubscribe = db.collection("posts").doc(postId).collection("comments").orderBy('timestamp', 'desc').onSnapshot(snapshot => {
                setComments(snapshot.docs.map(doc =>
                    doc.data()
                ))
            });
        }

        return () => {
            unsubscribe();
        }
    }, [postId]);

    const postComment = (e) => {
        e.preventDefault();

        db.collection("posts").doc(postId).collection("comments").add({
            comment: comment,
            username: user.displayName,
            timestamp: firebase.firestore.FieldValue.serverTimestamp(),
        });

        setComment('')
    };

    return (
        <div className="post">
            <div className="post__header">
                <Avatar
                    className="post__avatar"
                    alt=''
                    src={photoURL}
                />
                <h4> {username} </h4>
            </div>

            <img className="post__image" src={imageUrl}></img>

            <h5 className="post__capture"><strong>{username}</strong> {caption}</h5>

            <div className="post__comments">
                {comments.map((comment) => (
                    <p>
                        <strong>{comment.username}</strong> {comment.comment}
                    </p>
                ))}
            </div>

            <Form onSubmit={postComment} className="post__input">
                <Form.Control
                    className="post__input__comment"
                    type="text"
                    placeholder="Enter your comment..."
                    value={comment}
                    onChange={e => setComment(e.target.value)} />
                <Button className="post__btn" disabled={!comment || !user} type="submit" onClick={postComment}>
                    Post
                </Button>
            </Form>

        </div>
    )
}

export default Post