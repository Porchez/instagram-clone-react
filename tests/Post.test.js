import React from 'react';
import { shallow } from 'enzyme';
import Post from '../src/Post';

describe('<Post /> Test Suite', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Post />)
  })

  it('<Post /> match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})