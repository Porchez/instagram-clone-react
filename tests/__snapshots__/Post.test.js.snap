// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Test Suite <App /> match snapshot 1`] = `"<div class=\\"app\\"><div class=\\"header\\"><div class=\\"container container__header\\"><img class=\\"header__image\\" src=\\"https://www.instagram.com/static/images/web/mobile_nav_type_logo.png/735145cfe0a4.png\\" alt=\\"\\"/><button type=\\"button\\" class=\\"btn btn-link btn-sm\\">Log In</button></div></div><div class=\\"container container__post\\"><div><div></div></div></div></div>"`;

exports[`<Post /> Test Suite <Post /> match snapshot 1`] = `"<div class=\\"post\\"><div class=\\"post__header\\"><div class=\\"MuiAvatar-root MuiAvatar-circle post__avatar MuiAvatar-colorDefault\\"><svg class=\\"MuiSvgIcon-root MuiAvatar-fallback\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z\\"></path></svg></div><h4>  </h4></div><img class=\\"post__image\\"/><h5 class=\\"post__capture\\"><strong></strong> </h5><div class=\\"post__comments\\"></div><form class=\\"post__input\\"><input type=\\"text\\" placeholder=\\"Enter your comment...\\" value=\\"\\" class=\\"post__input__comment form-control\\"/><button disabled=\\"\\" type=\\"submit\\" class=\\"post__btn btn btn-primary\\">Post</button></form></div>"`;
